src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs

```diff
@@ -0,0 +1,126 @@
+                var symbol = semanticModel.GetDeclaredSymbol(diagnosticNode, cancellationToken);
+                Debug.Assert(symbol != null);
+
+                // Get symbol declarations to be removed.
+                var declarationService = document.GetLanguageService<ISymbolDeclarationService>();
+                foreach (var declReference in declarationService.GetDeclarations(symbol))
+                {
+                    var node = await declReference.GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                    declarators.Add(node);
+
+                    // For fields, the declaration node is the variable declarator.
+                    // We also track the ancestor FieldDeclarationSyntax which may declare more then one field.
+                    if (symbol.Kind == SymbolKind.Field)
+                    {
+                        var fieldDeclarator = node.FirstAncestorOrSelf<TFieldDeclarationSyntax>();
+                        fieldDeclarators.Add(fieldDeclarator);
+                    }
+                }
+            }
+
+            // If all the fields declared within a field declaration are unused,
+            // we can remove the entire field declaration instead of individual variable declarators.
+            if (fieldDeclarators.Count > 0)
+            {
+                AdjustDeclarators(fieldDeclarators, declarators);
+            }
+
+            // Remove all the symbol declarator nodes.
+            foreach (var declarator in declarators)
+            {
+                editor.RemoveNode(declarator);
+            }
+        }
+
+        /// <summary>
+        /// If all the <paramref name="childDeclarators"/> are contained in <paramref name="declarators"/>,
+        /// the removes the <paramref name="childDeclarators"/> from <paramref name="declarators"/>, and
+        /// adds the <paramref name="parentDeclaration"/> to the <paramref name="declarators"/>.
+        /// </summary>
+        protected void AdjustChildDeclarators(SyntaxNode parentDeclaration, IEnumerable<SyntaxNode> childDeclarators, HashSet<SyntaxNode> declarators)
+        {
+            if(declarators.Contains(parentDeclaration))
+            {
+                Debug.Assert(childDeclarators.All(c => !declarators.Contains(c)));
+                return;
+            }
+
+            var declaratorsContainsAllChildren = true;
+            foreach (var childDeclarator in childDeclarators)
+            {
+                if (!declarators.Contains(childDeclarator))
+                {
+                    declaratorsContainsAllChildren = false;
+                    break;
+                }
+            }
+
+            if (declaratorsContainsAllChildren)
+            {
+                // Remove the entire parent declaration instead of individual child declarators within it.
+                declarators.Add(parentDeclaration);
+                declarators.RemoveAll(childDeclarators);
+            }
+        }
```

Comment:
ok.  given this helper, it's unclear to me why there was the override in the C# subclass...